agent/eni/watcher_linux.go

```diff
@@ -50,8 +46,9 @@ type UdevWatcher struct {
 }
 
 // New is used to return an instance of the UdevWatcher struct
-func New(ctx context.Context, udevwrap udevwrapper.Udev) *UdevWatcher {
-	return _new(ctx, netlinkwrapper.New(), udevwrap, eniStateManager.New())
+func New(ctx context.Context, udevwrap udevwrapper.Udev, state dockerstate.TaskEngineState, taskEngine engine.TaskEngine) *UdevWatcher {
```

Comment:
None of the methods in this file are covered by unit tests. Please add test coverage for these methods.